stable/database/files/nuosm

```diff
@@ -748,9 +761,16 @@ trace "executing nuodocker to start SM"
 nuodocker_flags=()
 [ -n "$NUODB_DEBUG" ] && nuodocker_flags+=("--debug")
 
+journal_flags=()
+if [ "$SEPARATE_JOURNAL" == "true" ]; then
+  journal_flags+=("--journal-dir")
+  journal_flags+=("${JOURNAL_DIR}")
+  log "journal-dir=${JOURNAL_DIR}"
+fi
+
 # expects NUOCMD_API_SERVER to be set.
 if [ -n "${NUODB_OPTIONS}" ] ; then
-    exec nuodocker "${nuodocker_flags[@]}" start sm --archive-dir "${DB_DIR}" --dba-user "${DB_USER}" --dba-password "${DB_PASSWORD}" --db-name "${DB_NAME}" --options "${NUODB_OPTIONS}" "$@"
+    exec nuodocker "${nuodocker_flags[@]}" start sm --archive-dir "${DB_DIR}" "${journal_flags[@]}" --dba-user "${DB_USER}" --dba-password "${DB_PASSWORD}" --db-name "${DB_NAME}" --options "${NUODB_OPTIONS}" "$@"
 else
-    exec nuodocker "${nuodocker_flags[@]}" start sm --archive-dir "${DB_DIR}" --dba-user "${DB_USER}" --dba-password "${DB_PASSWORD}" --db-name "${DB_NAME}" "$@"
+    exec nuodocker "${nuodocker_flags[@]}" start sm --archive-dir "${DB_DIR}" "${journal_flags[@]}" --dba-user "${DB_USER}" --dba-password "${DB_PASSWORD}" --db-name "${DB_NAME}" "$@"
```

Comment:
I really wish `nuosm` would just pass all arguments verbatim to `nuodocker start sm` rather than duplicating the command-line processing and injecting its own defaults. Is it possible to just make `--journal-dir <the journal directory>` one of the opaque arguments that is passed verbatim? That would probably require converting some the Bash logic to Helm logic, but it would preserve the behavior of not specifying any explicit journal directory, allowing the SM to choose `${DB_DIR}/journal`.

Reply:
Actually, it looks like `journal_flags` would be empty, so we are relying on the internal default.